src/details/ArborX_DetailsTreeConstruction.hpp

```diff
@@ -204,16 +186,13 @@ class GenerateHierarchy
     // In this case, if the Morton indices are the same, we want to compare is.
     // We also want the result in this situation to always be less than any
     // Morton comparison. Thus, we add LLONG_MIN to it.
-    auto x = _sorted_morton_codes(i) ^ _sorted_morton_codes(i + 1);
-    if (x != 0)
-    {
-      // When using 63 bits for Morton codes, the LLONG_MAX is actually a valid
-      // code. As we want the return statement above to return a value always
-      // greater than anything here, we downshift by 1.
-      return x - 1;
-    }
+    auto const x = _sorted_morton_codes(i) ^ _sorted_morton_codes(i + 1);
 
-    return LLONG_MIN + (i ^ (i + 1));
+    return x + (!x) * (min_value + (i ^ (i + 1))) - 1;
```

Comment:
You accidentally introduced a subtlety here that one need to remember about. It is still correct, but now also depends on the fact that `i ^ (i+1) > 0`. Otherwise, you would have subtracted -1 from `min_value`.

Actually, I'm starting to dislike the whole `x + !x` thing. I am not sure it benefits the performance anymore, and it looks complex. We could try to
```c++
     auto const x = _sorted_morton_codes(i) ^ _sorted_morton_codes(i + 1);
     if (x)
       return x - 1;
     return min_value + i ^ (i + 1);
```
Does not have to be a part of this PR, though.